### ISSUE TITLE
SSLCertVerificationError, --no-verify option is not honored

### ISSUE THREAD

**Issue by Freedzone** [0xInfection/XSRFProbe#42]
#### Describe the bug
The `--no-verify` option is not checked when making requests in the **core/main.py** and in the **core/inputin.py**
```
requests.get(web)
```

#### Command You Used
First, I proxied the terminal through the ZAP proxy 
```
export https_proxy="http://localhost:8082"
```
Then run
```
xsrfprobe --no-verify -vv --malicious --cookie 'session=gGYCowmFLMcxL1dnloxmiVqN7qO5ILak' -u https://ac171f651e1e1a0e80c94df100620040.web-security-academy.net/email
```

#### Full Stack Trace Error
```
 Traceback (most recent call last):
  File "/usr/lib/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib/python3.7/http/client.py", line 1252, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1298, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1247, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/usr/lib/python3.7/http/client.py", line 1422, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python3.7/ssl.py", line 423, in wrap_socket
    session=session
  File "/usr/lib/python3.7/ssl.py", line 870, in _create
    self.do_handshake()
  File "/usr/lib/python3.7/ssl.py", line 1139, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1076)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./start.py", line 15, in <module>
    xsrfprobe.startEngine()
  File "/home/nyzen/tools/XSRFProbe/xsrfprobe/xsrfprobe.py", line 14, in startEngine
    main.Engine()  # start the Scanner Engine ;)
  File "/home/nyzen/tools/XSRFProbe/xsrfprobe/core/main.py", line 98, in Engine
    resp2.open(init1)  # Make request as User1
  File "/usr/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 1360, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1076)>
```

#### Potential cause or fix
**Inputin.py** should be easy: 
```   
requests.get(web, verify=VERIFY_CERT)
```
**main.py** is more [difficult](https://www.tutorialexample.com/best-practice-to-openerdirector-ignore-ssl-verification-in-python-3-x-python-web-crawler-tutorial/). I did not come up with something short and graceful so far
```python
    Cookie0 = http.cookiejar.CookieJar()  # First as User1
    Cookie1 = http.cookiejar.CookieJar()  # Then as User2
    if not config.VERIFY_CERT:
        context=ssl._create_unverified_context()
        sslHandler = urllib.request.HTTPSHandler(context=context)
        resp1 = build_opener(HTTPCookieProcessor(Cookie0), sslHandler)
        resp2 = build_opener(HTTPCookieProcessor(Cookie1), sslHandler) 
        # resp1.add_handler(sslHandler)  -----> this won't work unfortunately, because there will be multiple HTTPSHandlers
        # resp2.add_handler(sslHandler)
    else:
        resp1 = build_opener(HTTPCookieProcessor(Cookie0))
        resp2 = build_opener(HTTPCookieProcessor(Cookie1)) 
```

#### Environment:
- OS: [Kali Rolling 2019.4]
- Python version: [v3.7.5]

#### Some Questions

- [x] I am using the latest version of XSFRProbe.
- [x] I installed the dependecies using `pip3` instead of `pip`.
- [x] I have read the documentation before submitting this issue.
- [x] I have checked the other issues to see if someone reported this before.

**Comment by 0xInfection** [comment]
Hi there,

This issue has been reported earlier as #36, but somehow I kept overlooking those initial requests. The initial requests were put so that just an initial verification for the endpoints is working fine for multiple subsequent requests, so I thought they didn't require much fine tuning. Turns out they were the greatest troublemakers in the entire code.

Also, thank you for pointing out a possible fix too! I've updated the code and pushed them to the [`dev`](https://github.com/0xInfection/XSRFProbe/tree/dev) branch. Please test and lmk. :)

Thank you once again!

Cheers.

**Comment by 0xInfection** [comment]
Hey @Pitenager,

Was this the same bug you were facing? Do lmk if the fix works for you (and sorry for overlooking the obvious). :)

**Comment by Freedzone** [comment]
@0xInfection fix works for me, thanks 👍

**Comment by 0xInfection** [comment]
Fixed in #41 via d5d8853.

**Comment by Pitenager** [comment]
Yeah @0xInfection it also works for me now.

Thank you for the continuous support and quick fix.

Cheers !
